### PR TITLE
Cr 3703 subscription

### PR DESCRIPTION
**Pull request by doronafota**

### CODE REVIEW

File: tests/knowledge/context/azure/test_subscription.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+from tests.knowledge.context.azure_context_test import AzureNoCloudAccountContextTest
+from tests.knowledge.context.test_context_annotation import context, TestOptions
+
+
+class TestSubscription(AzureNoCloudAccountContextTest):
+
+    def get_component(self):
+        return 'subscription'
+
+    # CR-3703 not succeeded create new subscruption
```
Comment (Imanuel-Mizrahi) [inline tests/knowledge/context/azure/test_subscription.py:12]: I would change the comment a bit, something like:
"unable to create new subscription using IaC, probably lack of permissions"
or something like it.

Comment (doronafota) [reply tests/knowledge/context/azure/test_subscription.py:12]: fixed

File: cloudrail/knowledge/context/azure/resources/subscription/azure_subscription.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    def get_keys(self) -> List[str]:
+        return [self.get_id()]
+
+    def get_name(self) -> str:
+        return self.subscription_name
+
+    def get_type(self, is_plural: bool = False) -> str:
+        return 'Subscription' + ('s' if is_plural else '')
+
+    def to_drift_detection_object(self) -> dict:
```
[the remark this answers is not in the document]
Comment (doronafota) [reply cloudrail/knowledge/context/azure/resources/subscription/azure_subscription.py:40]: we don't check at all drift in this case
Is it necessary?

File: cloudrail/knowledge/context/azure/azure_relations_assigner.py [hunk 3]
```diff
@@ -161,3 +165,14 @@ def _assign_application_security_group_to_ip_config(network_interface: AzureNetw
             for asg_id in ip_config.application_security_groups_ids:
                 asg = ResourceInvalidator.get_by_id(asgs, asg_id, True, network_interface)
                 ip_config.application_security_groups.append(asg)
+
+    @staticmethod
+    def _assign_monitor_activity_log_alert_to_subscription(subscription: AzureSubscription, monitor_activity_log_alert: AliasesDict[AzureMonitorActivityLogAlert]):
+        def get_monitors():
+            monitors = []
```
[the remark this answers is not in the document]
Comment (doronafota) [reply cloudrail/knowledge/context/azure/azure_relations_assigner.py:172]: fixed
changed subscription.monitor_activity_alert_log_list getting a List

**Review by Imanuel-Mizrahi — APPROVED** [review]

**Review by Viperain — APPROVED** [review]

**Pull request merged**